lib/rmt/repo_manager.rb

```diff
@@ -1,115 +1,81 @@
-      mirroring_enabled = (@options[:enable]) ? true : false
-
-      if (@options[:repository])
-        change_repository_mirroring(mirroring_enabled, @options[:repository])
-      elsif (@options[:product])
-        change_product_mirroring(mirroring_enabled, @options[:identifier], @options[:version], @options[:arch])
-      else
-        raise 'Need to specify which repository to enable/disable!'
-      end
-
-      @out.puts "#{@affected_repos} repo(s) successfully #{mirroring_enabled ? 'enabled' : 'disabled'}"
+  desc 'enable TARGET', 'Enable a repository or product'
+  option 'exclude-optional', aliases: '-x', default: true, type: :boolean
+  def enable(target)
+    repo_id = Integer(target, 10) rescue nil
+    if repo_id
+      change_repository_mirroring(true, repo_id)
     else
-      list_repositories
+      identifier, version, arch = target.split('/')
+      change_product_mirroring(true, identifier, version, arch, exclude_optional: options['exclude-optional'])
     end
   end
 
+  desc 'version', 'Show version'
+  def version
+    puts RMT::VERSION
+  end
+
+
   protected
 
   def change_repository_mirroring(mirroring_enabled, repository_id)
     repository = Repository.find(repository_id)
     repository.mirroring_enabled = mirroring_enabled
     repository.save!
-    @affected_repos += 1
+    puts "Repo successfully #{mirroring_enabled ? 'enabled' : 'disabled'}."
   end
 
-  def change_product_mirroring(mirroring_enabled, identifier, version, arch = nil)
+  def change_product_mirroring(mirroring_enabled, identifier, version, arch, options = {})
     conditions = { identifier: identifier, version: version }
     conditions[:arch] = arch if arch
+    repo_count = 0
 
     products = Product.where(conditions).all
     products.each do |product|
-      conditions = {}
-      conditions[:enabled] = true if (@options[:exclude_optional])
-      @affected_repos += product.repositories.where(conditions).update_all(mirroring_enabled: mirroring_enabled)
+      conditions = { mirroring_enabled: !mirroring_enabled } # to only update the repos which need change
+      conditions[:enabled] = true if (options[:exclude_optional])
+      repo_count += product.repositories.where(conditions).update_all(mirroring_enabled: mirroring_enabled)
     end
+    puts "#{repo_count} repo(s) successfully #{mirroring_enabled ? 'enabled' : 'disabled'}."
   end
 
-  def list_repositories
+  def list_repositories(scope: :enabled)
     conditions = {}
-    conditions[:mirroring_enabled] = true unless (@options[:all])
+    conditions[:mirroring_enabled] = true unless (scope == :all)
 
     rows = []
     repositories = Repository.where(conditions)
     repositories.all.each do |repository|
       rows << [ repository.id, repository.name, repository.distro_target, repository.description, repository.mirroring_enabled ]
     end
 
-    @out.puts Terminal::Table.new headings: %w[ID Name Target Description Mirror?], rows: rows
```

Comment:
> Rubocop's complaining about using `puts`

Yes, but is the Rails/Output cop complaining. Does not really apply for a CLI tool. I think I'll just disable this cop for these files :relaxed: 

> also might come handy for capturing output in tests

AFAIR you can capture STDOUT with RSpec for a while now. See https://stackoverflow.com/a/16507680/3449673

> Or even rather replace puts it with logger.

Somethink like [this](https://stackoverflow.com/a/6407200/3449673) could do.
I'll give it a try.